tests/news/tests.py

```diff
@@ -0,0 +1,155 @@
+from contextlib import contextmanager
+from pathlib import Path
+
+from django.contrib.admin.tests import AdminSeleniumTestCase
+from django.contrib.auth.models import User
+from django.test import modify_settings, override_settings
+from django.test.selenium import ChangeWindowSize, screenshot_cases
+from django.urls import reverse
+from django.utils.text import slugify
+
+from .models import Article
+
+SIZES = ["docs_size"]
+
+
+@modify_settings(INSTALLED_APPS={"append": "django.contrib.flatpages"})
+@override_settings(ROOT_URLCONF="news.urls")
+class SeleniumTests(AdminSeleniumTestCase):
+    available_apps = AdminSeleniumTestCase.available_apps + [
+        "django.contrib.flatpages",
+        "news",
+    ]
+
+    def setUp(self):
+        self.superuser = User.objects.create_superuser(
+            username="admin",
+            password="secret",
+            email="admin@example.com",
+            first_name="Admin",
+            last_name="Super",
+        )
+        self.admin_login(username="admin", password="secret")
+
+    @contextmanager
+    def docs_size(self):
+        with ChangeWindowSize(800, 800, self.selenium):
+            yield
+
+    def take_screenshot(self, name):
+        path = Path.cwd().parent / "docs" / "ref" / "contrib" / "admin" / "_images"
+        assert path.exists()
+        self.selenium.save_screenshot(path / f"{name}.png")
+
+    def hide_nav_sidebar(self):
+        """Hide the navigation sidebar if it's open."""
+        from selenium.webdriver.common.by import By
+
+        if self.selenium.find_element(By.ID, "nav-sidebar").is_displayed():
+            self.selenium.find_element(By.CSS_SELECTOR, "#toggle-nav-sidebar").click()
+        self.selenium.execute_script(
+            'document.getElementById("toggle-nav-sidebar").style.display = "none";'
+        )
+
+    def change_header_display(self, display):
+        self.selenium.execute_script(
+            f'document.getElementById("header").style.display = "{display}";'
+        )
+        self.selenium.execute_script(
+            f'document.getElementsByTagName("nav")[0].style.display = "{display}";'
+        )
+
+    @screenshot_cases(SIZES)
+    def test_user_actions(self):
+        from selenium.webdriver import ActionChains
+        from selenium.webdriver.common.by import By
+
+        users = [
+            ("Adrian", "Holovaty", False),
+            ("Admin", "User", True),
+            ("Jacob", "Kaplan-Moss", True),
+            ("Simon", "Willison", False),
+        ]
```

Comment:
(side topic to be ignored in the context of this PR) I know these are here because they are the creators of Django, I wonder if everyone knows this, or they just look like a list of male English-y names 🤔